_source/_posts/2022-10-01-spring-vault.md

```diff
@@ -0,0 +1,639 @@
+
+{% img blog/spring-vault/vault-web-ui.png alt:"Vault web UI" width:"800" %}{: .center-image }
+
+Next, go to **Policies** and **Create ACL policy**. Create a `vault-demo-app-policy` with the following capabilities:
+
+```json
+path "secret/data/vault-demo-app" {
+  capabilities = [ "read" ]
+}
+
+path "secret/data/vault-demo-app,dev" {
+  capabilities = [ "read" ]
+}
+
+path "secret/data/application" {
+  capabilities = [ "read" ]
+}
+
+path "secret/data/application,dev" {
+  capabilities = [ "read" ]
+}
+```
+
+All the paths above will be requested by the config server to provide configuration for the `vault-demo-app` when it starts with the `dev` profile active.
+
+{% img blog/spring-vault/vault-policy.png alt:"Vault policy section" width:"800" %}{: .center-image }
+
+Now, go back to the container command line, and create a token with the `vault-demo-app-policy`.
+
+```shell
+vault token create -policy=vault-demo-app-policy
+```
+
+```shell
+WARNING! The following warnings were returned from Vault:
+
+  * Endpoint ignored these unrecognized parameters: [display_name entity_alias
+  explicit_max_ttl num_uses period policies renewable ttl type]
+
+Key                  Value
+---                  -----
+token                hvs.CAESIKd9pYyc9xesiqmwvepdGiHlPEB53By...
+token_accessor       2pzRXmGhxChobbMtqeM5zZzM
+token_duration       768h
+token_renewable      true
+token_policies       ["default" "vault-demo-app-policy"]
+identity_policies    []
+policies             ["default" "vault-demo-app-policy"]
+```
+
+**NOTE**: I could not find documentation about the warning _Endpoint ignored these unrecognized parameters_. It seems `vault CLI` is sending default parameters not required by the target API in this case. The command equivalent API call can be displayed using the `-output-curl-string` flag after the subcommand, for example:
+
+```bash
+vault token create -output-curl-string -policy=vault-demo-app-policy
+```
+
+```bash
+curl -X POST -H "X-Vault-Token: $(vault print token)" -H "X-Vault-Request: true" -d '{"policies":["vault-demo-app-policy"],"ttl":"0s","explicit_max_ttl":"0s","period":"0s","display_name":"","num_uses":0,"renewable":true,"type":"service","entity_alias":""}' http://127.0.0.1:8200/v1/auth/token/create
+```
+
+You are now ready to update the config server. In the `vault-config-server` project, edit `src/main/resource/application.yml` to add Vault as the config backend:
+
+```yml
+server:
+  port: 8888
+
+spring:
+  profiles:
+    active: vault
+  cloud:
+    config:
+      server:
+        vault:
+          host: 127.0.0.1
+          port: 8200
+          kvVersion: 2
+logging:
+  level:
+    root: TRACE
+```
+
+Note that the logging level is set to TRACE to see the interaction between the server and Vault. Restart the `vault-config-server`.
+
+```shell
+./mvnw spring-boot:run
+```
+
+You should see the logs below if the server was configured correctly:
```

Comment:
There's so much logging, it's really tough to dig in and find the message below. Is it possible to just enable trace logging for Spring Cloud Config?

Reply:
I fixed the log level.